scripts/erd/erd.py

```diff
@@ -0,0 +1,204 @@
+TABLE_TO_GROUP_MAP: dict[str, str] = {}
+for group, tables in GROUPINGS.items():
+    for table in tables:
+        TABLE_TO_GROUP_MAP[table] = group
+
+
+def introspect_sqla_model(mapper: Any, seen: set[str]) -> dict[str, Any]:
+    """
+    Introspects a SQLAlchemy model and returns a data structure that
+    can be pass to a jinja2 template for instance
+
+    Parameters:
+    -----------
+    mapper: SQLAlchemy model mapper
+    seen: set of model identifiers to avoid duplicates
+
+    Returns:
+    --------
+    Dict[str, Any]: data structure for jinja2 template
+    """
+    table_name = mapper.persist_selectable.name
+    model_info: dict[str, Any] = {
+        "class_name": mapper.class_.__name__,
+        "table_name": table_name,
+        "fields": [],
+        "relationships": [],
+    }
+    # Collect fields (columns) and their types
+    for column in mapper.columns:
+        field_info: dict[str, str] = {
+            "field_name": column.key,
+            "type": str(column.type),
+        }
+        model_info["fields"].append(field_info)
+
+    # Collect relationships and identify types
+    for attr, relationship in mapper.relationships.items():
+        related_table = relationship.mapper.persist_selectable.name
+        # Create a unique identifier for the relationship to avoid duplicates
+        relationship_id = "-".join(sorted([table_name, related_table]))
+
+        if relationship_id not in seen:
+            seen.add(relationship_id)
+            squiggle = "||--|{"
+            if relationship.direction.name == "MANYTOONE":
+                squiggle = "}|--||"
+
+            relationship_info: dict[str, str] = {
+                "relationship_name": attr,
+                "related_model": relationship.mapper.class_.__name__,
+                "type": relationship.direction.name,
+                "related_table": related_table,
+            }
+            # Identify many-to-many by checking for secondary table
+            if relationship.secondary is not None:
+                squiggle = "}|--|{"
+                relationship_info["type"] = "many-to-many"
+                relationship_info["secondary_table"] = relationship.secondary.name
+
+            relationship_info["squiggle"] = squiggle
+            model_info["relationships"].append(relationship_info)
+    return model_info
+
+
+def introspect_models() -> dict[str, list[dict[str, Any]]]:
+    """
+    Introspects SQLAlchemy models and returns a data structure that
+    can be pass to a jinja2 template for rendering an ERD.
+
+    Returns:
+    --------
+    Dict[str, List[Dict[str, Any]]]: data structure for jinja2 template
+    """
+    data: dict[str, list[dict[str, Any]]] = defaultdict(list)
+    seen_models: set[str] = set()
+    for model in db.Model.registry.mappers:
+        group_name = (
+            TABLE_TO_GROUP_MAP.get(model.mapper.persist_selectable.name)
+            or "Other Models"
```

Comment:
Should it be "other" (a bit of a junk drawer) or "Uncategorized Models" implying tech debt to be swept up?